activerecord/lib/active_record/connection_handling.rb

```diff
@@ -87,6 +91,19 @@ def connection
       retrieve_connection
     end
 
+    def specification_id=(value)
+      @specification_id = value
+    end
+
+    # Return the specification id from this class otherwise look it up
+    # in the parent.
+    def specification_id
+      unless defined?(@specification_id)
+        @specification_id = self == Base ? "primary" : superclass.specification_id
```

Comment:
i tried that, but it didnt work. I think because these are class methods. dunno much tbh.